src/Stratis.Bitcoin/Consensus/ConsensusManagerBehavior.cs

```diff
@@ -454,11 +449,25 @@ protected async Task<ConnectNewHeadersResult> PresentHeadersLockedAsync(List<Blo
             }
             catch (ConnectHeaderException)
             {
+                // This is typically thrown when the first header refers to a previous block hash that is not in
+                // the header tree currently. This is not regarded as bannable, as it could be a legitimate reorg.
                 this.logger.LogDebug("Unable to connect headers.");
-                this.cachedHeaders.Clear();
 
-                // Resync in case can't connect.
-                await this.ResyncAsync().ConfigureAwait(false);
+                if (this.CheckIfUnsolicitedFutureHeader(headers))
```

Comment:
I can see how this might cause issues with the integration tests as we are almost always going to advertise blocks ahead of consensus tip as we are able to mine "fast". So here we will almost always just log and not call re-sync (which might cause tests to timeout?).

Reply:
That is a good point, I'll investigate further

Reply:
Ah, it's actually way more subtle than that. The pre-created blockchain data is obviously from well in the past. So any header from a block mined on top of the premade chain is regarded as being an unsolicited future header, even though in the normal course of events it would not be, due to the low tip age.

@fassadlr

Reply:
Ah yes of course! Good job 👍